### ISSUE TITLE
Output property missing during initial preview when the input includes a nested unknown value

### ISSUE THREAD

**Issue by justinvp** [pulumi/pulumi#7854]
To reproduce, in a new empty directory run `pulumi new aws-typescript`.

Update `index.ts` with the following:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

// Use this resource's ARN for an unknown value.
const key = new aws.kms.Key("a");

// This bucket has no unknown values in its tags object; the tags are available in the policy pack at preview time as
// expected.
new aws.s3.Bucket("bucket-without-unknowns", {
    tags: {
        promptValue: "promptValue",
    }
});

// This bucket has an unknown value in its tags object. We would expect that the 'foo' tag value would be available at
// preview time, because it's provided as a prompt value; however, the entire object is treated as an unknown at preview
new aws.s3.Bucket("bucket-with-nested-unknown", {
    tags: {
        promptValue: "promptValue",
        unknown: key.arn
    }
});
```

Create a `policy` directory inside the project dir, `cd` into that directory and run `pulumi policy new aws-typescript`.

Update `index.ts` with the following:

```typescript
import * as aws from "@pulumi/aws";
import {PolicyPack, ReportViolation, StackValidationArgs} from "@pulumi/policy";

new PolicyPack("aws-typescript", {
    policies: [
        {
            name: "print-s3-props",
            description: "Print the props of S3 bucket policy resources",
            enforcementLevel: "mandatory",
            validateStack: async (args: StackValidationArgs, reportViolation: ReportViolation): Promise<void> => {
                args.resources.forEach((policyResource => {
                    if (policyResource.isType(aws.s3.Bucket)) {
                        console.log("Resource Name:", policyResource.name)
                        console.log("Resource Props:", policyResource.props)
                    }
                }))
            },
        },
    ],
});
```

In this example, there are two buckets, each of which has a `tags` argument. Each of the buckets has one tag that is a prompt value. One of the buckets has a tag with an unknown value and one of them does not. There is a policy pack that prints out the `props` of each policy resource. It is expected that the `tags` prop would be available on both buckets. However, the `tags` object containing an unknown value is not available at preview time.

Change back into the project directory and run:

```
pulumi preview --policy-pack ./policy
```

This results in:

```
Previewing update (dev)

     Type                 Name                        Plan       Info
 +   pulumi:pulumi:Stack  obj-previews-dev            create     16 messages
 +   ├─ aws:kms:Key       a                           create
 +   ├─ aws:s3:Bucket     bucket-without-unknowns     create
 +   └─ aws:s3:Bucket     bucket-with-nested-unknown  create


Diagnostics:
  pulumi:pulumi:Stack ((obj-previews-dev):
    Resource Name: bucket-without-unknowns
    Resource Props: {
      acl: 'private',
      bucket: 'bucket-without-unknowns-1690972',
      forceDestroy: false,
      id: '',
      tags: { promptValue: 'promptValue' },
      tagsAll: { promptValue: 'promptValue' }
    }
    Resource Name: bucket-with-nested-unknown
    Resource Props: {
      acl: 'private',
      bucket: 'bucket-with-nested-unknown-e3a8e08',
      forceDestroy: false,
      id: ''
    }
```

In summary, if an object input property contains an unknown value, the property is missing from the outputs, even if some of the keys in the object are provided as prompt values.

**Comment by tyler-dunkel** [comment]
@justinvp is this something that would be updated in the policy engine or is this before the policy engine

**Comment by Frassle** [comment]
Passing inputs in addition to outputs to stack policies would be an policy engine change. Changing the TF provider to not blank out all of the tags field just because one entry is unknown would be a tfbridge change.
